asm/load_store.go

```diff
@@ -223,3 +241,25 @@ func StoreXAdd(dst, src Register, size Size) Instruction {
 		Src:    src,
 	}
 }
+
+// LoadAcquire emits an atomic load with acquire semantics.
+func LoadAcquire(dst, src Register, offset int16, size Size) Instruction {
+	return Instruction{
+		OpCode:   StoreXAddOp(size), // We use Store here, for a Load, this is a quirk of the ISA
+		Dst:      dst,
+		Src:      src,
+		Offset:   offset,
+		Constant: int64(LoadAcquireAtomicFlag),
+	}
+}
+
+// StoreRelease emits an atomic store with release semantics.
+func StoreRelease(dst, src Register, offset int16, size Size) Instruction {
```

Comment:
I kept the descriptions on these brief for now. Not sure how much context we need to give the user about what these instructions do.

Reply:
Being brief is fine!